chemprop/cli/train.py

```diff
@@ -446,9 +449,9 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
 
 
 def process_train_args(args: Namespace) -> Namespace:
-    args.data_path = Path(args.data_path)
-
-    args.output_dir = Path(args.output_dir or Path.cwd() / args.data_path.stem)
+    args.output_dir = args.output_dir or Path("chemprop_training") / (
+        "_".join([args.data_path.stem, NOW])
+    )
```

Comment:
```suggestion
    if args.output_dir is None:
        args.output_dir =  Path(f"chemprop_training/{args.data_path.stem}_{NOW}")
```

Reply:
Using an f string like you did is better. 

Do you think using `if x is None` is better than `x or y`? Line 148 of predict.py is `args.output = args.output or Path(args.test_path.stem + "_preds.csv")`, so I'd like to either change both or neither to keep the style consistent.

Reply:
depends on whether a "falsy" `x` is a valid input. "Truthiness" and "falsiness" is a concept in programming when non-boolean datatypes can be coerced into booleans (see [here](https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false) for more details). In python, this can happen in a variety of places, such as `x or y` statement, which is syntatic sugar for `x if bool(x) else y` if `x` is not of type `bool`.  Writing the code this way shows us the possible perils with the shorthand `x or y`. If `x` is type of `str | None`, `bool(x)` will evaluate to `False` in both the case that `x` is `None` _**and**_ where `x` is the empty string `""` (`bool("") == False`). If the empty string _is_ a valid input, then the shorthand form won't work because it will be coerced to `False` and you will end up taking `y`'s value.

Reply:
Thanks David, I didn't know this before. In this case both `args.output_dir` and `args.out` will be Path objects when created (using `type=Path` in `parser.add_argument()`) and Path objects are never falsy. So using `or` would work. But also I think using `if x is None` is faster to understand so that is what I will go with. 

Taking a look at the other place we use `or`, I wonder if any of these should be changed to `if x is None`:
* I don't think an empty list `[]` is valid input for `bond_types` so using `or` [here](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/featurizers/bond.py#L63) is good. 
* I think an empty list is valid for stereo in [bond featurizers](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/featurizers/bond.py#L69). It would in effect turn off the stereochemistry featurization. If so, this would need to be changed in another PR.